File: TypeScript/Codewars/src/resolutions/RGB-To-Hex-Conversion.ts
export function rgb(r: number, g: number, b: number): string {
	const rgb = [r, g, b];
	const output: string[] = [];

	rgb.map((value) => {
		output.push(convertDecInHex(value));
	});

	return output.join("");
}

function convertDecInHex(n: number): string {
	let tempValue = n;

	if (n < 0) tempValue = 0;

	if (n > 255) tempValue = 255;

	return tempValue.toString(16).padStart(2, "0").toUpperCase();
}

File: Csharp/CodeWars/src/Completed/RgbToHexConversion.cs
namespace CodeWars.Completed;

public class RgbToHexConversion
{
    public static string Rgb(int r, int g, int b)
    {
        List<int> rgb = [r, g, b];
        List<string> output = [];

        foreach (var item in rgb)
            output.Add(ConvertDecInHex(item));

        return string.Concat(output);
    }

    private static string ConvertDecInHex(int r)
    {
        var tempValue = r;

        if (r < 0)
            tempValue = 0;

        if (r > 255)
            tempValue = 255;

        return Convert.ToString(tempValue, 16)
                 .PadLeft(2, '0')
                 .ToUpper();
    }
}